docs/content/concepts/resources.mdx

```diff
@@ -5,7 +5,24 @@ description: Resources enable you to separate graph logic from environment, and
 
 # Resources
 
-Resources provide a way to manage dependencies to external components and share implementations across multiple ops in a job.
+Resources provide a way to manage dependencies to external components and share implementations across multiple [ops](/concepts/ops-jobs-graphs/ops) or [assets](/concepts/assets/software-defined-assets) in a [job](/concepts/ops-jobs-graphs/jobs-graphs).
+
+Using resources, you could:
```

Comment:
Snippets like "Using resources, you could" and "why use resources" make me think that the content that follows is going to help me understand the situations and needs I have that resources would be appropriate to use in.  I think answering those is a worthy goal, but I worry that the current content we have doesn't really meet it.

E.g. I think "Bind a set of resources and other environment information to a job so that those resources can be available to the ops within that job" is really a sentence whose aim is to explain what a resource is, not what you can do with it.

E.g. I think the bullets like "Pluggable" etc. answer the question "what's cool about resources?", but don't exactly answer the question "why use resources"?

Reply:
That's fair. Perhaps a better way to phrase `Using resources, you could` is something like `Resources allow you to` and then keep the bulleted list that follows.

As for `why use resources?`, perhaps change this first sentence to `What are the benefits of using resources?` and then re-phrase the bulleted list as needed. Thoughts?

Reply:
I do think those are improvements in wording, but they have mostly the same meaning as the prior wordings, so I think the issues I raised are still there.

E.g. I think that "Provide access to features of the execution environment to ops" fits well under "Resources allow you to", but I still don't think "Bind a set of resources and other environment information to a job so that those resources can be available to the ops within that job" does. The latter is explaining _how_ you use resources, not what they allow you to do.

Reply:
Sorry for being difficult on this! I can take a stab at rewording it if it would be helpful.

Reply:
It's okay! If you have a moment to take a shot at this, it would be really helpful. I'll keep thinking in the meantime.

Reply:
For context, my version is heavily lifted from this paragraph in the [existing docs](https://docs.dagster.io/concepts/resources#overview):

> You can use resources to provide access to features of the execution environment to ops. You can bind a set of resources (and other environment information) to a job so that those resources can be available to the ops within that job. You can construct different jobs for the same graph, each with different resources, to represent the execution environments that your graph will be run within.

For me this was an exercise in improving readability, so I'm definitely not attached to maintaining a specific approach or wording by including this. Just letting you see where it came from!

The same is true for the `Pluggable, etc` bit ([see here](https://docs.dagster.io/concepts/resources#why-use-resources))

Reply:
I took a stab at this. What do you think?
---
**Resources** are objects that can be shared across the implementations of multiple assets and ops, and that can be plugged in after defining those ops and assets.

Resources are typically used to model external components that assets and ops interact with - e.g. a resource might be a connection to a data warehouse like Snowflake or a service like Slack.

Why use resources?

- **Plug in different implementations in different environments** - if you have a heavy external dependency that you want to use in production, but avoid using in testing, you can accomplish this by providing different resources in each environment. For more information about this capability, check out [Separating Business Logic from Environments](/concepts/testing#separating-business-logic-from-environments).
- **Share configuration across multiple ops or assets** - resources are configurable and shared, so you can supply configuration in one place instead of configuring the ops and assets individually.
- **Share implementations across multiple ops or assets** - when multiple ops access the same external services, resources provide a standard way to structure your code to share the implementations.